vmdb/app/models/ems_refresh/parsers/openstack_infra.rb

```diff
@@ -75,8 +75,14 @@ def cloud_ems_hosts_attributes
         return hosts_attributes unless clouds
 
         clouds.each do |cloud_ems|
-          connection = cloud_ems.connect
-          compute_hosts  = connection.hosts.select { |x| x.service_name == "compute" }
+          connection, compute_hosts = nil
+          next unless cloud_ems_safe_request("cloud ems connection failed for [#{cloud_ems.inspect}]") do
```

Comment:
This feels like ems' [with_provider_connection](https://github.com/ManageIQ/manageiq/blob/951bbd71e52254ee58d02ae93e7a0e780a1f8980/vmdb/app/models/ext_management_system.rb#L190).  I'm not a fan of blindly rescuing all exceptions via `Exception`.  I'd prefer something to use `with_provider_connection` as it already wraps the connect with a log message:
`$log.info("MIQ(#{self.class.name}.with_provider_connection) Connecting through #{self.class.name}: [#{self.name}]")`

```ruby
begin
  cloud_ems.with_provider_connection do |connection|
    compute_hosts  = connection.hosts.select { |x| x.service_name == "compute" }
  end
rescue SomeConnectException, CloudEmsHostListFailureException  # figure out what exceptions it should be... OR at worse, rescue StandardError.
  next
end